src/fastmcp/server/auth/providers/transparent_proxy.py

```diff
@@ -0,0 +1,511 @@
+    AuthorizeError,
+    RefreshToken,
+    TokenError,
+)
+from mcp.shared.auth import OAuthClientInformationFull, OAuthToken
+from pydantic import AnyHttpUrl, SecretStr, AnyUrl, ValidationError
+
+from fastmcp.server.auth.auth import (
+    ClientRegistrationOptions,
+    OAuthProvider,
+    RevocationOptions,
+)
+from fastmcp.server.auth.providers.bearer import BearerAuthProvider
+from fastmcp.utilities.logging import get_logger
+
+logger = get_logger(__name__)
+
+DEFAULT_ACCESS_TOKEN_EXPIRY_SECONDS: Final[int] = 60 * 60  # 1h
+DEFAULT_AUTH_CODE_EXPIRY_SECONDS: Final[int] = 5 * 60  # 5m
+
+
+class TransparentOAuthProxyProvider(OAuthProvider):
+    """An OAuthProvider implementation that proxies most endpoints to an upstream
+    Authorization Server while locally implementing Dynamic Client Registration.
+    """
+
+    def __init__(
+        self,
+        *,
+        # Upstream details
+        upstream_authorization_endpoint: str,
+        upstream_token_endpoint: str,
+        upstream_jwks_uri: str,
+        upstream_client_id: str,
+        upstream_client_secret: str,
+        # FastMCP (this server) public details
+        issuer_url: AnyHttpUrl | str,
+        service_documentation_url: AnyHttpUrl | str | None = None,
+        client_registration_options: ClientRegistrationOptions | None = None,
+        revocation_options: RevocationOptions | None = None,
+        required_scopes: list[str] | None = None,
+    ) -> None:
+        super().__init__(
+            issuer_url=issuer_url,
+            service_documentation_url=service_documentation_url,
+            client_registration_options=client_registration_options
+            or ClientRegistrationOptions(enabled=True),
+            revocation_options=revocation_options,
+            required_scopes=required_scopes,
+        )
+
+        self._upstream_authorization_endpoint = upstream_authorization_endpoint
+        self._upstream_token_endpoint = upstream_token_endpoint
+        self._upstream_jwks_uri = upstream_jwks_uri
+
+        self._upstream_client_id = upstream_client_id
+        self._upstream_client_secret = SecretStr(upstream_client_secret)
+
+        # Local state (DCR + auth codes)
+        self._clients: dict[str, OAuthClientInformationFull] = {}
+        self._auth_codes: dict[str, AuthorizationCode] = {}
+        self._access_tokens: dict[str, AccessToken] = {}
+        self._refresh_tokens: dict[str, RefreshToken] = {}
+        # Map relations for rotation / revocation
+        self._access_to_refresh: dict[str, str] = {}
+        self._refresh_to_access: dict[str, str] = {}
+
+        # Validate signature and expiration using upstream JWKS, but do NOT enforce
+        # issuer matching because the tokens were issued by the upstream provider
+        # (e.g., "https://developer.api.autodesk.com"), not by this proxy.
+        self._bearer_validator = BearerAuthProvider(
+            jwks_uri=self._upstream_jwks_uri,
+            issuer=None,
```

Comment:
Since we have the `issuer_url` from the upstream Authorization Server as parameter for the initialization of the proxy, why not adding it here?